feat: robots directive, full robots tag support (#12998)
shopware/platform: UPGRADE-6.8.md

# 6.8.0.0

## Removal of string parameter in `DomainRuleStruct` constructor

> The deprecated string parameter in the `Shopware\Storefront\Page\Robots\Struct\DomainRuleStruct` constructor will be removed. Use `Shopware\Storefront\Page\Robots\Parser\RobotsDirectiveParser::parse()` to create a `ParsedRobots` object instead.

Reviewer: ```suggestion
The deprecated string parameter in the `Shopware\Storefront\Page\Robots\Struct\DomainRuleStruct` constructor was removed. 
Use `Shopware\Storefront\Page\Robots\Parser\RobotsDirectiveParser::parse()` to create a `ParsedRobots` object instead.
```

# 6.8.0.0

> ## Removal of string parameter in `DomainRuleStruct` constructor

Reviewer: we should wait until https://github.com/shopware/shopware/pull/13346 is merged with the new structure

 {
     // snip

> 

Reviewer: merge foo here and all changes below
